### ISSUE TITLE
Grunt tasks

### ISSUE THREAD

**Issue by ahmadalfy** [drdrsh/html-tree-builder#1]
Currently, following the steps in the readme doesn't work. The dist directory doesn't get created till you use `grunt dist`.

I would suggest making the `dist` the default task and update the readme. The user should run this sequence:

```
grunt

grunt serve
```

What do you think?

**Comment by drdrsh** [comment]
Sorry for that.

My workflow is that I have "serve" run from "dev" not "dist" , this is done manually by modifying Gruntfile.js, I need this to have all my scripts in readable form. dist is not meant for development.

Serve doesn't have "target" support so I have to manually switch between "dist" and "dev" manually.

**Comment by drdrsh** [comment]
It doesn't make much sense to "serve" dist because dist should be served by a proper web server. Still, I need to be able to look at the "dist" before I commit it to make sure that all uglification and compression didn't mess up my code.

**Comment by ahmadalfy** [comment]
True. What I regularly see is that `serve` act on a separate directory like `tmp`. Perhapse this is something we can do?

**Comment by drdrsh** [comment]
So I would have tmp, dev and dist?
I think having tmp makes sense If I do any "pre-processing" for my code which I don't, in this case I don't see the benefit of having a tmp.

Ideally I should have the freedom to serve dist or dev if I so please, the task shouldn't constrain me.

**Comment by ahmadalfy** [comment]
Yup, it is ideal for preprocessing and so it isn't needed for now. I think `serve` should serve from dev not dist. If someone wants to try dist he can use a web server or even open the file in the browser directly. It will work with the file protocol perfectly, you're not doing any ajax calls.

**Comment by drdrsh** [comment]
Agreed.

Still, I friggin' hate this. Why do I have to use an external server to serve from the "dist" directory? Why does the developer baby-sit me and tell me what I can and cannot do?  I don't need this feature to have grunt serve my files to the public, who would ever do that! _sigh_
